Rpackage/R/fitAndScore.R

```diff
@@ -0,0 +1,191 @@
+    
+    fold_split_idx$train <- rbind(fold_split_idx$train, this_row_train)
+    fold_split_idx$test <- rbind(fold_split_idx$test, this_row_test)
+  }
+  
+  ### end row
+  this_row_train <- seq(from = (as.integer((l/n_folds)+1)), to = l, by = 1)
+  this_row_test <- seq(from = 1, to = as.integer((l/n_folds)), by = 1)
+  
+  fold_split_idx$train <- rbind(fold_split_idx$train, this_row_train)
+  fold_split_idx$test <- rbind(fold_split_idx$test, this_row_test)
+  
+  ### fold trees
+  fold_tree <- NULL
+  for(i in 1:n_folds){
+    fold_tree[[i]] <- mmit(target.mat[fold_split_idx$train[i,],], feature.mat[fold_split_idx$train[i,],],  
+                                       maxdepth = as.numeric(parameters$maxdepth), margin = as.numeric(parameters$margin), 
+                                       loss = loss, min_sample = as.numeric(parameters$min_sample))
+  }
+  
+  ### master tree
+  master_tree <- mmit(target.mat, feature.mat,  
+                      maxdepth = as.numeric(parameters$maxdepth), margin = as.numeric(parameters$margin), 
+                      loss = loss, min_sample = as.numeric(parameters$min_sample))
+  
+  
+  ### if pruning
+  if(pruning){
+    # Get the pruned master and cross-validation trees
+    master_data <- pruning(master_tree)
+    master_alphas <- master_data[, 1]
+    master_pruned_trees <- master_data[, -1]
+    
+    fold_alphas <- NULL
+    fold_prune_trees <- NULL
+    for(i in 1 : n_folds){
+      fold_data <- pruning(fold_tree[[i]])
+      fold_alphas[[i]] <- fold_data[, 1] 
+      fold_prune_trees[[i]] <- fold_data[, -1]
+    }
+    
+    ### alphas list should not contain repeating alpha
+    master_alphas <- master_alphas[!duplicated(master_alphas)]
+    fold_alphas <- fold_alphas[!duplicated(fold_alphas)]
+    
+    
+    # Compute the test risk for all pruned trees of each fold
+    alpha_path_score <- NULL
+    for(i in 1 : n_folds){
+      for(j in 1 : length(fold_prune_trees[[i]][,1])){
+        ### convert pruned tree list to partynode
+        node <- fold_prune_trees[[i]][[j]]
+        
+        fit <- fitted_node(node, feature.mat[fold_split_idx$test[i,],])
+        n <- nodeapply(node, ids = fit, info_node)
+        ###prediction of test data
+        prediction <- matrix(unlist(n), nrow = length(n), byrow = T)[,1]
+        ###error calc
+        fold_test_scores <- scorer(target.mat[fold_split_idx$test[i,],], prediction)
+        
+        ### creating a dataframe with init alpha, final alpha, score value
+        if(j < (length(fold_alphas[[i]]) - 1)){
+          alpha_path_score <- rbind(alpha_path_score, c(fold_alphas[[i]][j], fold_alphas[[i]][j+1], fold_test_scores))
+        }
+        else{
+          alpha_path_score <- rbind(alpha_path_score, c(fold_alphas[[i]][j], Inf, fold_test_scores))
+        }
+        colnames(alpha_path_score) <- c("init alpha", " final alpha", "score")
+        alpha_path_score <- as.data.frame(alpha_path_score)
```

Comment:
There is something that I don't understand. You are using one single alpha_path_score for all folds, but adding one row for each pruned tree of each fold. Then you are setting the colnames each time and converting it to a dataframe (each time).

In alpha_path_score, you should be able to easily access the alpha path for each fold, separately.

Reply:
Also, you can't expect to have the same number of alphas for each fold.

Reply:
Live meeting: decided to use a list of dataframes.